oslokommune/okctl
# 22Q2-18 🐛 Fix delete app not regarding env

## Description

- Deletes overlay directory for the relevant cluster in all cases
- Introduces a check to see if there is 0 associated clusters, which if true,
    we delete the whole application root directory

## Motivation and Context


[22Q2-18](https://trello.com/c/0ZBgCAbZ)

## How to prove the effect of this PR?





### Test the case of no related clusters
1. Add a new application with `okctl apply application`
2. Run `okctl delete application` and verify that the application root
    directory is deleted.

### Test the case of one or more related clusters
1. Add a new application with `okctl apply application`.
2. Run `cp -r infrastructure/<yourcluster>{,-mock}` and
    `cp -r infrastructure/applications/<app>/overlays/<yourcluster>{,-mock}`
	to emulate another associated cluster.
3. Run `okctl delete application` and verify that only the relevant overlays
    directory for the relevant cluster is deleted and that the application
	root directory is still intact.

## Additional info



## Types of changes



## Checklist:



- [ ] I have updated the release notes (for the next release).
